modules/cas_cache/layer_cache_management.c

```diff
@@ -1999,32 +2019,43 @@ int cache_mngt_set_cache_mode(const char *cache_name, size_t name_len,
 	if (result)
 		return result;
 
-	result = _cache_mngt_lock_sync(cache);
-	if (result) {
-		ocf_mngt_cache_put(cache);
-		return result;
+	old_mode = ocf_cache_get_mode(cache);
+	if (old_mode == mode) {
+		printk(KERN_INFO "%s is in defined cache mode already", cache_name);
```

Comment:
would be nice to repeat this check after acquiring management write lock - just to make sure cache mode did not change in between